.github/workflows/main.yml

```diff
@@ -0,0 +1,132 @@
+name: Tbots CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  formatting-check:
+    runs-on: ubuntu-18.04
+    steps:
+      # checks-out the repository under $GITHUB_WORKSPACE
+      - uses: actions/checkout@v2
+        
+      - name: Environment Setup
+        run: |
+          "${GITHUB_WORKSPACE}"/environment_setup/setup_software.sh
+      
+      - name: Check Formatting
+        run: |
+          ./formatting_scripts/check_formatting_ci.sh
+          
+  software-tests:
+    name: Software Tests
+    runs-on: ubuntu-18.04
+    steps:
+      # checks-out the repository under $GITHUB_WORKSPACE
+      - uses: actions/checkout@v2
+
+      - name: Environment Setup
+        run: |
+          "${GITHUB_WORKSPACE}"/environment_setup/setup_software.sh
+          
+      - name: Cache Bazel External Dependencies
+        uses: actions/cache@v2
```

Comment:
Do firmware targets not benefit from caching external? It also seems like it's caching test results and not just external dependencies. Can we cache just external targets?

Reply:
changed to just external targets